sdk/core/Azure.Core.Experimental/api/Azure.Core.Experimental.net461.cs

```diff
@@ -191,17 +192,77 @@ public partial struct DynamicJsonOptions
         public string Name { get { throw null; } }
         public Azure.Core.Dynamic.DynamicJson Value { get { throw null; } }
     }
+    public abstract partial class ObjectDocument : System.IDisposable
+    {
+        protected ObjectDocument() { }
+        public abstract T As<T>(object element);
+        public abstract void Dispose();
+        public abstract Azure.Core.Dynamic.ObjectElement GetIndexElement(object element, int index);
+        public abstract bool HasValue(object element);
+        public abstract void Set(object element, object value);
+        public abstract Azure.Core.Dynamic.ObjectElement SetProperty(object element, string name, object value);
+        public abstract string? ToString(object element);
```

Comment:
Why is this ToString (returning nullable) as opposed to TryGetString?

Reply:
It's because [object.ToString()](https://learn.microsoft.com/en-us/dotnet/api/system.object.tostring?view=net-8.0) returns `string?`.  We also have TryGetString() to get the value as a string.

Reply:
Sorry, I asked the question very poorly. What I meant is that TryGetString seems to be doing exactly what this method does, and so why do we need this method in addition to TryGetString?

Reply:
Ah, I see.  I think we want the different methods - ToString() will return the value as a JSON string in all cases, for example, if it is a Number type, it would return e.g. `"5.1"`, or if it is an Object type, it would return `{"foo": "hi"}`.  TryGetString() will return false if the value is not a string type.  But looking at the [JsonElement API](https://learn.microsoft.com/en-us/dotnet/api/system.text.json.jsonelement?view=net-8.0), I see that they to not have a TryGetString() - I guess that is because you can check the ValueKind property and if it is a string, ToString() will give you the string value.  We can follow their pattern if we introduce ValueKind.

Reply:
Also, I am not sure how callers of the abstraction (which can not be JSON) should think about this API, i.e. how would anybody use it successfully without knowing what this will return when called on an object.

Reply:
This is intended to just be a pass through to calling ToString() on the element.  I guess since ToString() is already on object, this is not needed.  Ok, removing.